### DISCUSSION TITLE
Method to offset frames after import

### DISCUSSION THREAD

**Discussion by asuter** [Autodesk/maya-usd#3423] [Q&A] [answered]
Hiya,

Is there a way to offset the animation of a cache within Maya once the USD has been imported? For example, my cache has animation from frame 1 to 100, could I make that happen on the timeline from frames 51 to 150?

**Comment by jcoelho88** [comment, marked as answer]
Hi @asuter,

From what I gather if its imported back to Maya data then there will be a bunch of blendShape nodes that will contain the keys.

Run this on a python tab in the script editor to select the blendShape nodes (assuming your import is the only thing in the scene)

```python
cmds.select(cmds.ls(type="blendShape"))
```
You should see the keys on the timeline, using the timeline you can select and shift the keys.

Just a note: Depending on how many blendShape nodes and how many keys it could take a minute.

Hope this helps.

Cheers,

Jason Coelho

**Reply by asuter** [reply]
Hi Jason,

Thanks for answering! That does look like it would work great for shape deformations, but I should have mentioned I'm looking to offset the joint animation in time. I have a USDSkel cache that loads into Maya 2023 great (with maya-usd 0.18.0), but I haven't found a connected node driving the time or anything like that yet. 

Thanks again,
Alex

**Reply by asuter** [reply]
Sorry, I was dumb. I had to do exactly what you said, but for animCurve nodes. That does the trick. Thank you.

**Reply by jcoelho88** [reply]
Glad you got it working!
